src/scss/src/foundation/_mixins.scss

```diff
@@ -1,16 +1,169 @@
+    @media (min-width: map-get($breakpoints, "tablet")) {
+        @content;
+    }
+}
+
+@mixin desktop {
+    @media (min-width: map-get($breakpoints, "desktop")) {
+        @content;
+    }
+}
+
+@mixin mobile {
+    @media (min-width: map-get($breakpoints, "mobile")) {
+        @content;
+    }
+}
+
+/*=============================================
+=                 Spacings                    =
+=============================================*/
+
+/**
+ * use it set up margin or padding 
+ * 
+ * @include spacing(margin, 'lg', 'xxs') - will set up margin vertical for 32px and horizontal for 8px 
+ */
+
+@mixin spacing($property, $values...) {
+    $valid-properties: margin, padding;
+
+    @if not list.index($valid-properties, $property) {
+        @error "#{$property} is not a valid property. Expected one of #{$valid-properties}.";
+    }
+
+    $propertyValues: ();
 
+    @each $value in $values {
+        $spacingValue: map-get($spacing, $value);
+        $propertyValues: append($propertyValues, $spacingValue);
+    }
 
+    #{$property}: $propertyValues;
+}
+
+/*=============================================
+=              Containers sizings            =
+=============================================*/
+
+/**
+ * defines width and height in one go
+ * 
+ * @include box(1rem, 2rem) => width: 1rem;  height: 2rem
+ */
+
+@mixin box($width, $height: $width) {
+    width: $width;
+    height: $height;
+}
+
+/*=============================================
+=               FLEX/GRID Toolkit            =
+=============================================*/
+
+/**
+ * Setup for flexbox container
+ *
+ * example usage: 
+ * @include flexbox(column, space-between)
+ * @include flexbox($justify: center)
+ *
+ */
+
+@mixin flexbox($direction: row, $justify: center, $align: center, $wrap: nowrap, $gap: normal) {
+    display: flex;
+    flex-direction: $direction;
+    justify-content: $justify;
+    align-items: $align;
+    flex-wrap: $wrap;
+    gap: $gap;
+}
+
+/**
+ * Setup for grid container
+ *    
+ * @include grid-container(1fr 1fr, 30px, start, center);
+ *
+ */
+
+@mixin grid($columns: auto, $rows: auto, $gap: 20px, $justify: center, $align: center) {
+    display: grid;
+    grid-template-columns: $columns;
+    grid-template-rows: $rows;
+    gap: $gap;
+    justify-items: $justify;
+    align-items: $align;
+}
+
+/*=============================================
+=            Text manipulations            =
+=============================================*/
+
+/**
+ * This mixin is used to truncate text that overflows its container.
+ * 
+ * To truncate text with ellipsis: @include text-truncate;
+ * 
+ * To truncate text with a custom string: @include text-truncate('...');
+ * 
+ * To clip the text without any ellipsis: @include text-truncate('clip');
+ * 
+ */
+
+@mixin text-truncate($overflow: ellipsis) {
+    overflow: hidden;
+    text-overflow: $overflow; // values are: clip, ellipsis, or a string
+    white-space: nowrap;
+}
```

Comment:
I suggest adding `text-clamp` mixin to limit number of lines displayed in paragragh